backend/src/controllers/groups-controller.ts

```diff
@@ -93,8 +115,22 @@ export const updateName = async (req: Request, res: Response) =>{
 }
 
 //Deletes group by ID
-export const deleteGroup = async (req: Request, res: Response) => {
+export const deleteGroup = async (req: CustomRequest, res: Response) => {
     const groupID: string = req.params._id;
+    //Checks if UserId exists
+    if(!req.token?.userId){
+        return res.status(422).json({error: "UserId does not exist"});
+    }
+    const groupOwner = req.body.userId;
+    const findGroupOwnerById = Group.find({ userId: groupOwner });
```

Comment:
(Repeat) Check if the user owns the group using `req.token.userId` and not `groupOwner`